streams/piping/then-interception.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+if (self.importScripts) {
+  self.importScripts('/resources/testharness.js');
+  self.importScripts('../resources/test-utils.js');
+  self.importScripts('../resources/recording-streams.js');
+}
+
+function interceptThen() {
```

Comment:
Seems a bit simpler to just make then() throw? The test will then fail if it's accessed.

Reply:
Yes, but less fun. "I can make pipeTo throw" is far less compelling than "I can make pipeTo write arbitrary data". Please indulge my desire to make this test exciting.